test/generators/suspenders/prerequisites_generator_test.rb

```diff
@@ -16,23 +16,32 @@ class PrerequisitesGeneratorTest < Rails::Generators::TestCase
           run_generator
 
           assert_file app_root(".node-version") do |file|
-            assert_match(/20\.0\.0/, file)
+            assert_match(/^20\.0\.0$/, file)
           end
         end
       end
 
+      test "#node_version parses out the version number from an alphanumeric value" do
+        ClimateControl.modify NODE_VERSION: nil do
+          Object.any_instance.stubs(:`).returns("v1.7.4\n")
+          actual = Suspenders::Generators::PrerequisitesGenerator.new.node_version
+
+          assert_match(/^1\.7\.4$/, actual)
```

Comment:
@stevepolitodesign and I tried using `assert_equal` here, but the `\n` in the stub causes the test output to not show expected vs. actual if you introduce a regression.

This [Slack conversation](https://thoughtbot.slack.com/archives/CKCV8VD1N/p1729891370940189?thread_ts=1729876745.842449&cid=CKCV8VD1N) has more details.